src/Git/GitStatusListCtrl.cpp

```diff
@@ -4490,3 +4531,78 @@ ULONG CGitStatusListCtrl::GetGestureStatus(CPoint /*ptTouch*/)
 {
 	return 0;
 }
+
+void CGitStatusListCtrl::LoadChangelists()
+{
+	CString changelistFLP = g_Git.m_CurrentDir + L"\\.tgitchangelist";
+	if (!PathFileExists(changelistFLP))
+		return;
+
+	try
+	{
+		CStdioFile file(changelistFLP, CFile::typeText | CFile::modeRead | CFile::shareDenyWrite);
+		CString changelistName = GITSLC_IGNORECHANGELIST;
+		CString strLine;
+		while (file.ReadString(strLine))
+		{
+			strLine = strLine.Trim();
+			if (strLine.IsEmpty())
+				continue;
+
+			if ((strLine.Left(1) == L'<') && (strLine.Right(1) == L'>'))
+			{ //this is changelist name
+				changelistName = strLine.Mid(1, strLine.GetLength() - 2);
+				m_changelists.insert_or_assign(changelistName, 0);
+			}
+			else // this is git path
+				m_pathToChangelist.insert_or_assign(strLine, changelistName);
+		}
+		file.Close();
+	}
+	catch (CFileException* pE)
+	{
+		CTraceToOutputDebugString::Instance()(__FUNCTION__ ": CFileException loading changelists file\n");
+		pE->Delete();
+		MessageBox(L"Unable to load changelists", L"TortoiseGit", MB_ICONERROR);
+	}
+}
+
+void CGitStatusListCtrl::SaveChangelists()
+{
+	std::map<CString, std::set<CString>> changelistToPath;
+	for (auto it = m_pathToChangelist.cbegin(); it != m_pathToChangelist.cend(); ++it)
+	{
+		auto itChangelist = changelistToPath.find(it->second);
+		if (itChangelist == changelistToPath.end())
+		{
+			std::set<CString> paths;
+			paths.insert(it->first);
+			changelistToPath.insert(std::make_pair(it->second, paths));
+		}
+		else
+			itChangelist->second.insert(it->first);
+	}
+
+	try
+	{
+		CStdioFile file(g_Git.m_CurrentDir + L"\\.tgitchangelist", CFile::typeText | CFile::modeCreate | CFile::modeWrite | CFile::shareDenyWrite);
+		for (auto itChangelist = changelistToPath.cbegin(); itChangelist != changelistToPath.cend(); ++itChangelist)
+		{
+			if (itChangelist != changelistToPath.cbegin())
+				file.WriteString(L"\n");
+
+			file.WriteString(L"<" + itChangelist->first + L">\n");
+			for (auto itPath = itChangelist->second.cbegin(); itPath != itChangelist->second.cend(); ++itPath)
+				file.WriteString(*itPath + L"\n");
+		}
+
+		file.Close();
+	}
+	catch (CFileException* pE)
+	{
+		CTraceToOutputDebugString::Instance()(__FUNCTION__ ": CFileException saving changelists file\n");

```

Comment:
As we also show a messagebox, so this debug output should not be necessary as it adds no further info